indexers/types.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 The Kube-burner Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package indexers
+
+// Types of indexers
+const (
+	// Elastic indexer that sends metrics to the configured ES instance
+	ElasticIndexer IndexerType = "elastic"
+	// OpenSearch indexer that sends metrics to the configured Search Instance
+	OpenSearchIndexer IndexerType = "opensearch"
+	// Local indexer that writes metrics to local directory
+	LocalIndexer IndexerType = "local"
+)
+
+// Indexer indexer interface
+type Indexer interface {
+	Index([]interface{}, IndexingOpts)
```

Comment:
@vishnuchalla agree with removing logging any implementation from here, let's return an error from this interface's method as well rather than printing the log

Reply:
Changed as suggested. Thanks!